cmd/dep/root_analyzer.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2016 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/hex"
+
+	"github.com/golang/dep"
+	fb "github.com/golang/dep/internal/feedback"
+	"github.com/golang/dep/internal/gps"
+)
+
+// importer handles importing configuration from other dependency managers into our format
+type importer interface {
+	Name() string
+	Import(path string, pr gps.ProjectRoot) (*dep.Manifest, *dep.Lock, error)
```

Comment:
I did it this way for a couple reasons:

1. I needed the importers to return a dep manifest and lock, not the gps interfaces. Otherwise I would have to cast it immediately back.
2. I didn't want the importers to implement functions that would go unused, `Analyzer.Info`.

So while they are similar, but I didn't see anything to be gained by reusing that interface.